rust-runtime/smithy-http/src/operation.rs

```diff
@@ -199,12 +211,68 @@ impl Request {
             .expect("a clone of a valid request should be a valid request");
         Some(Request {
             inner,
-            configuration: self.configuration.clone(),
+            properties: self.properties.clone(),
         })
     }
 
+    /// Consumes the operation `Request` and returns the underlying HTTP request and properties.
     pub fn into_parts(self) -> (http::Request<SdkBody>, Arc<Mutex<PropertyBag>>) {
-        (self.inner, self.configuration)
+        (self.inner, self.properties)
+    }
+}
+
+/// Operation response type that associates a property bag with an underlying HTTP response.
+/// This type represents the response in the Tower `Service` in middleware so that middleware
+/// can share information with each other via the properties.
+#[derive(Debug)]
+pub struct Response {
+    /// The underlying HTTP Response
+    inner: http::Response<SdkBody>,
+
+    /// Property bag of configuration options
+    ///
+    /// Middleware can read and write from the property bag and use its
+    /// contents to augment the request (see [`Response::augment`](Response::augment))
```

Comment:
I don't think this function actually exists?

Reply:
Good catch. Removed.